File: Frontend/src/app/classes/visita/visita-historico.ts
export enum EVisitaHistoricoTipo {
    
    AtividadeRealizada = 1,
    InteracaoCliente = 2,
    Chamada = 3,
    Todos = 99

}

export const VisitaHistoricoTipos = [
    { codigo: EVisitaHistoricoTipo.Todos, descricao: 'Todos', sigla: '' },
    { codigo: EVisitaHistoricoTipo.AtividadeRealizada, descricao: 'Atividade Realizada', sigla: 'ATIVIDADE' },
    { codigo: EVisitaHistoricoTipo.InteracaoCliente, descricao: 'Interação com Cliente', sigla: 'INTERACAO' },
    { codigo: EVisitaHistoricoTipo.Chamada, descricao: 'Chamada', sigla: 'CHAMADA' },
];

export class VisitaHistorico {

    usina: number = 0;
    numeroVisita: number = 0;
    anoVisita: number = 0;
    
    tipoHistorico: EVisitaHistoricoTipo = EVisitaHistoricoTipo.AtividadeRealizada;
    descricao: string = '';

    data: Date = new Date();

    hora: string = new Date().getHours().toString().padStart(2, '0') + ':' + new Date().getMinutes().toString().padStart(2, '0') + ':00';
    horaString: string = new Date().getHours().toString().padStart(2, '0') + new Date().getMinutes().toString().padStart(2, '0');

    dataRetorno: Date = null;

    horaRetorno: string = '';
    horaRetornoString: string = '';

    idCadastro: string = '';

}

File: Backend/TopSys.TopConWeb.Domain/Entities/Visitas/VisitaHistorico.cs
using System;
using TopSys.TopConWeb.Domain.Enums;

namespace TopSys.TopConWeb.Domain.Entities.Visitas
{
    public class VisitaHistorico
    {

        public Guid Id { get; set; }

        public int Usina { get; set; }
        public int NumeroVisita { get; set; }
        public int AnoVisita { get; set; }

        public string Tipo { get; set; } = "ATIVIDADE";
        public string Descricao { get; set; }

        public DateTime Data { get; set; }
        public TimeSpan Hora { get; set; }

        public DateTime? DataRetorno { get; set; }
        public TimeSpan? HoraRetorno { get; set; }

        public string IdCadastro { get; set; }

        public EVisitaHistoricoTipo TipoHistorico
        {
            get
            {
                switch (this.Tipo)
                {
                    case "ATIVIDADE":
                        return EVisitaHistoricoTipo.AtividadeRealizada;
                    case "INTERACAO":
                        return EVisitaHistoricoTipo.InteracaoCliente;
                    case "CHAMADA":
                        return EVisitaHistoricoTipo.Chamada;
                    default:
                        return EVisitaHistoricoTipo.AtividadeRealizada;
                }
            }

            set
            {
    
                if (value == EVisitaHistoricoTipo.AtividadeRealizada)
                    this.Tipo = "ATIVIDADE";

                if (value == EVisitaHistoricoTipo.InteracaoCliente)
                    this.Tipo = "INTERACAO";

                if (value == EVisitaHistoricoTipo.Chamada)
                    this.Tipo = "CHAMADA";

            }

        }

    }
}
